drivers/wireless/gs2200m.c

```diff
@@ -3071,6 +3086,34 @@ static void gs2200m_irq_worker(FAR void *arg)
 
   t = gs2200m_recv_pkt(dev, pkt_dat);
 
+  if (true == gs2200m_disassociate_flag)
+    {
+      /* Disassociate recovery */
+
+      wlwarn("=== receive DISASSOCIATE\n");
+      dev->valid_cid_bits = 0;
+
+      gs2200m_disassociate(dev);
+      do
+        {
+          while (gs2200m_recv_pkt(dev, pkt_dat) != TYPE_TIMEOUT)
+            {
```

Comment:
Do we have more packets to receive?

Reply:
If some socket programs send packets before disassociated, many "ERROR: INVALID INPUT" response is received in this timing.

Reply:
Hmm, so you mean that we might receive "ERROR: INVALID INPUT"  messages AFTER we received "Disassociation Event".  So here, we need to call gs2200m_recv_pkt() to discard such messages UNTIL timeout happens.
Is my understanding correct?

Reply:
Yes,  that's right.

Reply:
I checked this situation. When gs2200m is disassociated, gs2200m resets all connections. But driver don't catch this condition immediately and keep sending and retrying bulk packet for a while. So. gs2200m respond "ERROR: INVALID INPUT" text message in this timing.

Reply:
OK, I understand.